### PR TITLE
Fix Delete-by-Query with Shield

### LINKED ISSUE #14527: delete-by-query failing with Shield enabled
On Found we are testing the delete-by-query plugin in 2.0.0. I have enabled the plugin and when I issue a delete via _query, I get the following response.

```
{"error":{"root_cause":[{"type":"security_exception","reason":"action [indices:data/read/search] is unauthorized for user []"}],"type":"security_exception","reason":"action [indices:data/read/search] is unauthorized for user []"},"status":403}
```

### PR DESCRIPTION
**Pull request by tlrx**
closes #14527

### CODE REVIEW

**Comment by tlrx** [comment]
@javanna Can you have a look please? Thanks

File: plugins/delete-by-query/src/main/java/org/elasticsearch/action/deletebyquery/TransportDeleteByQueryAction.java [hunk 1]
```diff
@@ -155,7 +150,11 @@ public void onFailure(Throwable e) {
         void executeScroll(final String scrollId) {
             try {
                 logger.trace("executing scroll request [{}]", scrollId);
-                scrollAction.execute(new SearchScrollRequest(scrollId).scroll(request.scroll()), new ActionListener<SearchResponse>() {
+                SearchScrollRequest searchScrollRequest = new SearchScrollRequest();
+                searchScrollRequest.copyContextAndHeadersFrom(request);
```
Comment (javanna) [inline plugins/delete-by-query/src/main/java/org/elasticsearch/action/deletebyquery/TransportDeleteByQueryAction.java]: can you instead add a new constructor to SearchScrollRequest that accepts the original request like we do elsewhere?

File: plugins/delete-by-query/src/main/java/org/elasticsearch/action/deletebyquery/TransportDeleteByQueryAction.java [hunk 2]
```diff
@@ -283,7 +283,10 @@ void finishHim(final String scrollId, boolean scrollTimedOut, Throwable failure)
                 }
 
                 if (Strings.hasText(scrollId)) {
-                    client.prepareClearScroll().addScrollId(scrollId).execute(new ActionListener<ClearScrollResponse>() {
+                    ClearScrollRequest clearScrollRequest = new ClearScrollRequest();
```
Comment (javanna) [inline plugins/delete-by-query/src/main/java/org/elasticsearch/action/deletebyquery/TransportDeleteByQueryAction.java]: same as above

**Comment by javanna** [comment]
left two minor comments, LGTM otherwise, thanks @tlrx

File: plugins/delete-by-query/src/main/java/org/elasticsearch/action/deletebyquery/TransportDeleteByQueryAction.java (hunk 2 above)
Comment (tlrx) [inline plugins/delete-by-query/src/main/java/org/elasticsearch/action/deletebyquery/TransportDeleteByQueryAction.java]: Should I do it for `BulkRequest` too? (line 200)

Comment (javanna) [inline plugins/delete-by-query/src/main/java/org/elasticsearch/action/deletebyquery/TransportDeleteByQueryAction.java]: yea I missed that, I think it makes sense to do it there too.

File: plugins/delete-by-query/src/main/java/org/elasticsearch/action/deletebyquery/TransportDeleteByQueryAction.java (hunk 1 above)
Comment (tlrx) [inline plugins/delete-by-query/src/main/java/org/elasticsearch/action/deletebyquery/TransportDeleteByQueryAction.java]: sure

**Comment by javanna** [comment]
LGTM

**Comment by tlrx** [comment]
@javanna Thanks!

**Pull request merged**